src/bitcoin.cpp

```diff
@@ -0,0 +1,207 @@
+#else
+#include <unistd.h>
+#endif
+
+extern const std::function<std::string(const char*)> G_TRANSLATION_FUN = nullptr;
+
+static constexpr auto HELP_USAGE = R"(Usage: %1$s [OPTIONS] COMMAND...
+
+Commands (run help command for more information):
+  {gui,daemon,rpc,wallet,test,help}
+
+Options:
+  -m, --multiprocess     Run multiprocess binaries bitcoin-node, bitcoin-gui.\n"
+  -M, --monolothic       Run monolithic binaries bitcoind, bitcoin-qt. (Default behavior)\n"
+  -v, --version          Show version information
+  -h, --help             Show this help message
+)";
+
+static constexpr auto HELP_COMMANDS = R"(Command overview:
+
+%1$s gui [ARGS]     Start GUI, equivalent to running 'bitcoin-qt [ARGS]' or 'bitcoin-gui [ARGS]'.
+%1$s daemon [ARGS]  Start daemon, equivalent to running 'bitcoind [ARGS]' or 'bitcoin-node [ARGS]'.
+%1$s rpc [ARGS]     Call RPC method, equivalent to running 'bitcoin-cli -named [ARGS]'.
+%1$s wallet [ARGS]  Call wallet command, equivalent to running 'bitcoin-wallet [ARGS]'.
+%1$s tx [ARGS]      Manipulate hex-encoded transactions, equivalent to running 'bitcoin-tx [ARGS]'.
+%1$s test [ARGS]    Run unit tests, equivalent to running 'test_bitcoin [ARGS]'.
+%1$s help           Show this help message.
+)";
+
+struct CommandLine {
+    bool use_multiprocess{false};
+    bool show_version{false};
+    bool show_help{false};
+    std::string_view command;
+    std::vector<std::string_view> args;
+};
+
+CommandLine ParseCommandLine(int argc, char* argv[]);
+void ExecCommand(const std::vector<std::string>& args, std::string_view argv0);
+
+int main(int argc, char* argv[])
+{
+    try {
+        CommandLine cmd{ParseCommandLine(argc, argv)};
+        if (cmd.show_version) {
+            std::cout << FormatParagraph(LicenseInfo());
+        }
+        if (cmd.show_help || cmd.command.empty()) {
+            tfm::format(std::cout, HELP_USAGE, argv[0]);
+        }
+
+        std::vector<std::string> args;
+        if (cmd.command == "gui") {
+            args.emplace_back(cmd.use_multiprocess ? "qt/bitcoin-gui" : "qt/bitcoin-qt");
+        } else if (cmd.command == "daemon") {
+            args.emplace_back(cmd.use_multiprocess ? "bitcoin-node" : "bitcoind");
+        } else if (cmd.command == "rpc") {
+            args.emplace_back("bitcoin-cli");
+            args.emplace_back("-named");
+        } else if (cmd.command == "wallet") {
+            args.emplace_back("bitcoin-wallet");
+        } else if (cmd.command == "tx") {
+            args.emplace_back("bitcoin-tx");
+        } else if (cmd.command == "test") {
+            args.emplace_back("test/test_bitcoin");
+        } else if (cmd.command == "help") {
+            tfm::format(std::cout, HELP_COMMANDS, argv[0]);
+        } else if (cmd.command == "mine") { // undocumented, used by tests
+            args.emplace_back("bitcoin-mine");
+        } else if (cmd.command == "util") { // undocumented, used by tests
+            args.emplace_back("bitcoin-util");
```

Comment:
> I think it would be better to expose a bitcoin grind command that doesn't require callers to know about underlying binaries

Ok, that makes sense.